extensions/amp-font/0.1/amp-font.js

```diff
@@ -0,0 +1,304 @@
+      };
+    });
+  }
+
+
+  /**
+   * Make the browser that don't support font loading events to download the
+   * custom font by creating an element (with text) not visible on the viewport.
+   * font download is detected by comparing the elements height and width with
+   * measurements taken in step1.
+   * @private
+   */
+  loadFont_() {
+    var haswidthChanged = false;
+    var hasHeightChanged = false;
+    var ncustomFontFamily_ = normalizeFontFamily_(this.customFontFamily_);
+    DEFAULT_FONTS_.some(font => {
+      if(this.hasTimedOut_) {
+        this.onFontLoadError_();
+        return true;
+      }
+      let nfont = normalizeFontFamily_(font);
+      let defaultFontElementId = '-amp-font-'+ ncustomFontFamily_ + '-' + nfont;
+      let element = document.getElementById(defaultFontElementId);
+      let currentFontFamily = element.style.fontFamily;
+      if (currentFontFamily.indexOf(this.customFontFamily_) == -1) {
+        element.style.fontFamily =
+            this.customFontFamily_ + ',' + currentFontFamily;
+      }
+      var defaultDimensions =
+          this.measurements_[defaultFontElementId];
+      haswidthChanged = (
+        Math.abs(
+          defaultDimensions.width - element./*OK*/offsetWidth) > TOLERANCE_);
+      hasHeightChanged = (
+        Math.abs(
+          defaultDimensions.height - element./*OK*/offsetHeight) > TOLERANCE_);
+      if (haswidthChanged || hasHeightChanged) {
+        return true;
+      }
+    });
+    if (haswidthChanged || hasHeightChanged) {
+      this.onFontLoadSuccess_();
+    } else if(this.hasTimedOut_) {
+      this.onFontLoadError_();
+      return true;
+    } else {
+      window.requestAnimationFrame(() => this.loadFont_());
+    }
+  }
+
+
+  /**
+   * @private
+   */
+  onFontLoadSuccess_() {
+    var addClassName = this.element.getAttribute('on-load-class') || '';
+    document.documentElement.classList.add(addClassName);
+    this.dispose();
+  }
+
+
+  /**
+   * @private
+   */
+  onFontLoadError_() {
+    var addClassName = this.element.getAttribute('on-error-class') || '';
+    document.documentElement.classList.add(addClassName);
+    this.dispose();
+  }
+
+
+  /**
+   * @private
+   */
+  dispose() {
+    var removeClassName = this.element.getAttribute('on-loading-class') || '';
+    document.documentElement.classList.remove(removeClassName);
+    this.measurements_ = {};
+    let containerElement = document.getElementById(
+       '-amp-font-container-'+normalizeFontFamily_(this.customFontFamily_));
+    if (containerElement) {
+      containerElement.parentNode.removeChild(containerElement);
+    }
+    if (this.timeoutHandle_) {
+      window.clearTimeout(this.timeoutHandle_);
+    }
+  }
+
+  /**
+   * Starts the timeout counter to not attempt loading the font after the
+   * timeout interva.
+   * @private
+   */
+  beginTimeout_() {
+    var timeoutAttribute = this.element.getAttribute('timeout');
+    var timeoutInMs;
+    timeoutInMs = (
```

Comment:
thats how some of our interfaces are (only supports ms), we'd want to be consistent across the board i think (whatever way we go)